src/QsFmt/App/Program.fs

```diff
@@ -29,13 +29,24 @@ let main args =
         let input = results.GetResult Input
         let source = if input = "-" then stdin.ReadToEnd() else File.ReadAllText input
 
-        match Formatter.format source with
-        | Ok result ->
-            printf "%s" result
-            0
+        match Formatter.parse source with
+        | Ok document ->
+            // Test whether there is data loss during parsing and unparsing
+            if Formatter.unparse document = source then
```

Comment:
It might be simpler to move this check inside the `Formatter.format` function. I think that you wouldn't need to expose new functions or types in that case.

Reply:
Agree. I have rolled back most changes in my last commit, i.e., I have added back `internal` labels and removed new exposing functions. Now, the check is within `Formatter.format` function.